nanoFramework.WebServer/HttpMultipartParser/MultipartFormDataParser.cs

```diff
@@ -0,0 +1,293 @@
+    public class MultipartFormDataParser
+    {
+        private const int defaultBufferSize = 4096;
+
+        private readonly bool _ignoreInvalidParts;
+        private readonly int _binaryBufferSize;
+        private string _boundary;
+        private byte[] _boundaryBinary;
+        private readonly Stream _stream;
+        private bool _readEndBoundary;
+        private readonly ArrayList _files = new();
+        private readonly ArrayList _parameters = new();
+
+        /// <summary>Initializes a new instance of the <see cref="MultipartFormDataParser" /> class</summary>
+        /// <param name="stream">The stream containing the multipart data.</param>
+        /// <param name="binaryBufferSize">The size of the buffer to use for parsing the multipart form data.</param>
+        /// <param name="ignoreInvalidParts">By default the parser will throw an exception if it encounters an invalid part. Set this to true to ignore invalid parts.</param>
+        public MultipartFormDataParser(Stream stream, int binaryBufferSize = defaultBufferSize, bool ignoreInvalidParts = false)
+        {
+            _stream = stream ?? throw new ArgumentNullException(nameof(stream));
+            _binaryBufferSize = binaryBufferSize;
+            _ignoreInvalidParts = ignoreInvalidParts;
+        }
+
+        /// <summary>Gets the mapping of parameters parsed files. The name of a given field maps to the parsed file data.</summary>
+        public FilePart[] Files => _files.ToArray(typeof(FilePart)) as FilePart[];
+
+        /// <summary>Gets the parameters. Several ParameterParts may share the same name.</summary>
+        public ParameterPart[] Parameters => _parameters.ToArray(typeof(ParameterPart)) as ParameterPart[];
+
+        /// <summary>Parse the stream into a new instance of the <see cref="MultipartFormDataParser" /> class</summary>
+        /// <param name="stream">The stream containing the multipart data.</param>
+        /// <param name="binaryBufferSize">The size of the buffer to use for parsing the multipart form data.</param>
+        /// <param name="ignoreInvalidParts">By default the parser will throw an exception if it encounters an invalid part. Set this to true to ignore invalid parts.</param>
+        /// <returns>A new instance of the <see cref="MultipartFormDataParser"/> class.</returns>
+        public static MultipartFormDataParser Parse(Stream stream, int binaryBufferSize = defaultBufferSize, bool ignoreInvalidParts = false)
+        {
+            var parser = new MultipartFormDataParser(stream, binaryBufferSize, ignoreInvalidParts);
+            parser.Run();
+            return parser;
+        }
+
+        private void Run()
+        {
+            var reader = new LineReader(_stream, _binaryBufferSize);
+
+            _boundary = DetectBoundary(reader);
+            _boundaryBinary = Encoding.UTF8.GetBytes(_boundary);
+
+            //we have read until we encountered the boundary so we should be at the first section => parse it!
```

Comment:
I know it's super minimal, but we're used to have a space after the double slash for comments. It appears in many places. IT's minimal, don't search for each of them, adjust the ones you'll touch.

Reply:
No problem ;-) Think I caught most of them on last commit.